### PR TITLE
feat(new server): Palworld

### LINKED ISSUE #4448: [Server Request]: Palworld
### Game server

Palworld

### Linux support

- [X] Yes

### Steam

Yes

### Steam appid

2394010

### Guides

[Official Documentation](https://tech.palworldgame.com/dedicated-server-guide#linux)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by PlayMTL**
# Description

This is the current standing of the implementation of the new server Palworld. Before merging, i would like to squash my commits :)

Fixes #4448 

## Type of change

-   [ ] Bug fix (a change which fixes an issue).
-   [ ] New feature (a change which adds functionality).
-   [X] New Server (new server added).
-   [ ] Refactor (restructures existing code).
-   [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

-   [X] This pull request links to an issue.
-   [X] This pull request uses the `develop` branch as its base.
-   [ ] This pull request subject follows the Conventional Commits standard.
-   [ ] This code follows the style guidelines of this project.
-   [X] I have performed a self-review of my code.
-   [X] I have checked that this code is commented where required.
-   [ ] I have provided a detailed enough description of this PR.
-   [ ] I have checked if documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.

-   User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
-   Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

### CODE REVIEW

File: lgsm/config-default/config-lgsm/pwserver/_default.cfg [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+## Predefined Parameters | https://docs.linuxgsm.com/configuration/start-parameters
+ip="0.0.0.0"
+port="8211"
+players="32"
+serverpassword=""
+adminpassword=""
+servername="LinuxGSM Palworld Server"
+
+## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
+## Game Server Docs | https://satisfactory.fandom.com/wiki/Dedicated_servers
```
Comment (h3o66) [inline lgsm/config-default/config-lgsm/pwserver/_default.cfg:20]: url from copied gameserver should be replace with the doc from the gameserver

File: lgsm/config-default/config-lgsm/pwserver/_default.cfg [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+startparameters="-port=${port} -players=${players} -useperfthreads -NoAsyncLoadingThread -UseMultithreadForDS -publicip ${ip} -publicport ${port} EpicApp=PalServer -servername=${servername} -serverpassword=${serverpassword} adminpassword=${adminpassword}"
```
Comment (h3o66) [inline lgsm/config-default/config-lgsm/pwserver/_default.cfg:21]: did you check if that some parmater work without `-` in front ? looks strange at first to me

**Review by h3o66 — CHANGES_REQUESTED** [review]

File: lgsm/config-default/config-lgsm/pwserver/_default.cfg (hunk 2 above)
Comment (dgibbs64) [reply lgsm/config-default/config-lgsm/pwserver/_default.cfg:21]: yeah looks like it should be `-EpicApp ` and `-adminpassword`

Comment (PlayMTL) [reply lgsm/config-default/config-lgsm/pwserver/_default.cfg:21]: yeah `-adminpassword` should be right, the `EpicApp` Settings should be okay, as it is...

Comment (dgibbs64) [reply lgsm/config-default/config-lgsm/pwserver/_default.cfg]: Looks like both work please add the - for consistancy

File: lgsm/modules/info_game.sh [hunk 3]
```diff
@@ -613,6 +613,17 @@ fn_info_game_pvr() {
 	servername="${servername:-"NOT SET"}"
 }
 
+# Config Type: ini
+# Parameters: false
+# Comment: ; or #
+# Example: ServerName=SERVERNAME
+# Filetype: cfg
+fn_info_game_pw() {
+	servername="${selfname:-"NOT SET"}"
```
Comment (dgibbs64) [inline lgsm/modules/info_game.sh:622]: this should be `servername="${servername:-"NOT SET"}"`

File: lgsm/modules/info_game.sh [hunk 4]
```diff
@@ -613,6 +613,17 @@ fn_info_game_pvr() {
...
+	port="${port:-"0"}"
+	maxplayers="${players:-"0"}"
```
Comment (dgibbs64) [inline lgsm/modules/info_game.sh:624]: maxplayers="${maxplayers:-"0"}"

File: lgsm/config-default/config-lgsm/pwserver/_default.cfg [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+stopmode="2"
+
+## Query mode
+# 1: session only
+# 2: gamedig (gsquery fallback)
+# 3: gamedig
+# 4: gsquery
+# 5: tcp
+querymode="1"
+querytype=""
```
Comment (dgibbs64) [inline lgsm/config-default/config-lgsm/pwserver/_default.cfg:144]: should be set to 1

**Comment by Thundernerd** [comment]
@PlayMTL don't forget to request a re-review once you've fixed everything that was requested, otherwise the reviewers might not see it.

File: lgsm/modules/info_game.sh (hunk 4 above)
Comment (dgibbs64) [reply lgsm/modules/info_game.sh:624]: maxplayers="${maxplayers:-"0"}"

**Comment by PlayMTL** [comment]
I think that's it now. There are some workarounds included, because the gameserver is a bit special...

File: lgsm/modules/info_messages.sh [hunk 6]
```diff
@@ -1269,6 +1269,14 @@ fn_info_messages_pvr() {
 	} | column -s $'\t' -t
 }
 
+fn_info_messages_pw() {
+	{
+		fn_port "header"
+		fn_port "Game" port udp
+		fn_port "Steam" steamport udp
```
[the remark this answers is not in the document]
Comment (PlayMTL) [reply lgsm/modules/info_messages.sh]: okay, i renamed the whole var to unknowport to keep it align

File: lgsm/config-default/config-lgsm/pwserver/_default.cfg [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+# Copy settings from here and use them in either:
+# common.cfg - applies settings to every instance.
+# [instance].cfg - applies settings to a specific instance.
+
+#### Game Server Settings ####
+
+## Predefined Parameters | https://docs.linuxgsm.com/configuration/start-parameters
+ip="0.0.0.0"
+port="8211"
+players="32"
```
[the remark this answers is not in the document]
Comment (PlayMTL) [reply lgsm/config-default/config-lgsm/pwserver/_default.cfg]: i decided to remove the startup arguments as much as possible. server now takes settings from the config file.

**Comment by AedDev** [comment]
Would be great if this PR makes it today! :)

**Comment by BlackDahlia313** [comment]
> the gameserver is a bit special...

no lie detected

**Comment by JamsRepos** [comment]
@dgibbs64 I hate to apply pressure but is there any plans of this being merged soon?

**Review by dgibbs64 — APPROVED** [review]

**Pull request merged**